server/src/main/java/org/candlepin/async/tasks/HealEntireOrgJob.java

```diff
@@ -0,0 +1,176 @@
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.async.tasks;
+
+import org.candlepin.async.ArgumentConversionException;
+import org.candlepin.async.AsyncJob;
+import org.candlepin.async.JobArguments;
+import org.candlepin.async.JobConfig;
+import org.candlepin.async.JobConfigValidationException;
+import org.candlepin.async.JobConstraints;
+import org.candlepin.async.JobExecutionContext;
+import org.candlepin.async.JobExecutionException;
+import org.candlepin.common.filter.LoggingFilter;
+import org.candlepin.controller.AutobindDisabledForOwnerException;
+import org.candlepin.controller.Entitler;
+import org.candlepin.model.Consumer;
+import org.candlepin.model.ConsumerCurator;
+import org.candlepin.model.Entitlement;
+import org.candlepin.model.Owner;
+import org.candlepin.model.OwnerCurator;
+import org.candlepin.resource.dto.AutobindData;
+
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+
+import org.jboss.resteasy.spi.BadRequestException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xnap.commons.i18n.I18n;
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * HealEntireOrgJob
+ */
+public class HealEntireOrgJob implements AsyncJob {
+
+    private static Logger log = LoggerFactory.getLogger(HealEntireOrgJob.class);
+    public static final String JOB_KEY = "HEAL_ENTIRE_ORG_JOB";
+    private static final String JOB_NAME = "heal entire org job";
+    public static final String OWNER_KEY = "org";
+    public static final String ENTITLE_DATE_KEY = "entitle_date";
+
+    private OwnerCurator ownerCurator;
+    private Entitler entitler;
+    private ConsumerCurator consumerCurator;
+    private I18n i18n;
+
+    @Inject
+    public HealEntireOrgJob(Entitler entitler, ConsumerCurator consumerCurator, OwnerCurator ownerCurator,
+        I18n i18n) {
+        this.entitler = entitler;
+        this.consumerCurator = consumerCurator;
+        this.ownerCurator = ownerCurator;
+        this.i18n = i18n;
+    }
+
+    @Override
+    public Object execute(JobExecutionContext context) throws JobExecutionException {
+        try {
+
+            final JobArguments arguments = context.getJobArguments();
+            final String ownerKey = arguments.getAsString(OWNER_KEY);
+            Owner owner = ownerCurator.getByKey(ownerKey);
+            if (owner.isAutobindDisabled() || owner.isContentAccessEnabled()) {
+                String caMessage = owner.isContentAccessEnabled() ?
+                    " because of the content access mode setting" : "";
+                throw new BadRequestException(
```

Comment:
Let's update this exception to something more suitable for the context (JobException), and ensure it's flagged as terminal so we don't attempt to retry the job.

Reply:
This exception catch by outer catch block and re-through as JobException. Still, do we need to change the exception type here?

Reply:
Probably should, as we're not in a request context here.

The rethrowing is fine-ish, we just need to be somewhat careful about whether or not the job should be retried. This particular failure is terminal, and the wrapper exception is set to non-terminal. Not a huge deal in the end; we'll just retry until the limit is hit, which will generate a bunch of log noise.

Reply:
I am creating new PR with all these changes. Thanks for suggestions.